File: 1342-Number-of-Steps-to-Reduce-a-Number-to-Zero/iterative.py
# https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/submissions/1041612029/

class Solution:
    def numberOfSteps(self, num: int) -> int:
        steps = 0
        while num != 0:
            if num % 2 != 0:
                num -= 1
            else:
                num /= 2
            steps += 1
        return steps

File: 0144-Binary-Tree-Preorder-Traversal/iterative.ts
// https://leetcode.com/problems/binary-tree-preorder-traversal/submissions/1177460153/

/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function preorderTraversal(root: TreeNode | null): number[] {
    if(root == null) return [];

    const res : number[] = [];
    const stack : TreeNode[] = [];

    stack.push(root);
    while(stack.length > 0) {
        const node = stack.pop();
        res.push(node.val);
        if(node.right) stack.push(node.right);
        if(node.left) stack.push(node.left);
    }
    
    return res;
};
